pkg/worker/verifier/producer_worker.go

```diff
@@ -399,8 +419,8 @@ func (pw *ProducerWorker) produceInner(n int64) (int64, []BadOffset, error) {
 		for o := range bad_offsets {
 			r = append(r, o)
 		}
-		if len(r) == 0 && pw.Status.FailedTransactions == 0 {
-			util.Die("No bad offsets or failed transactions but errored?")
+		if len(r) == 0 && pw.Status.Fails == 0 && pw.Status.FailedTransactions == 0 {
+			util.Die("No bad offsets or failed produces or transactions but errored?")
```

Comment:
@nvartolomei I've fixed this line we discussed, please could you have another look